### DISCUSSION TITLE
License

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#74] [General]
It would be good to add a license to the project. Maybe [MIT](https://choosealicense.com/licenses/mit/)?

**Comment by CEXT-Dan** [comment]
Not sure how that works yet since I’m using other open source, wxWindows is L-GPL

**Comment by gswifort** [comment]
The LGPL license doesn't seem to prohibit you from licensing derivative software under the MIT license if you don't modify the source software.

**Reply by CEXT-Dan** [reply]
Yeah it’s hard to understand.

```MIT License Compatibility
The MIT License is highly compatible with other permissive licenses. Including the BSD family of licenses. 
It is generally compatible with  GNU GPL group of licenses. 
However if you distribute the code that contains or is derivative of GNU GPL code the final project must of GPL compliant.
In other words any source code must of publicly available.
```


https://www.wxwidgets.org/about/licence/ is GPL with an exception 
```
2. The exception is that you may use, copy, link, modify and distribute
under your own terms, binary object code versions of works based on the
Library.
```

The way I read it, my project is not in binary form, and,  “if you distribute the code that **contains** or is derivative of GNU GPL code the final project must of GPL compliant”


I’m thinking I would need to use LGPL

**Comment by gswifort** [comment]
Some discussions on this topic:

https://opensource.stackexchange.com/questions/10537/mit-license-ok-if-dependent-package-uses-lgpl

https://softwareengineering.stackexchange.com/questions/141847/how-does-using-a-lgpl-gem-affect-my-mit-licensed-application

https://www.reddit.com/r/rust/comments/fevz37/want_to_use_lgpl_licensed_library_do_i_need_to/

**Comment by CEXT-Dan** [comment]
Pretty sure LGPL is going to be the best fit, it would be possible, (in theory) to go with MIT because I’m using dynamic linking.
While investigating exception 2 of the wxWidgets license, which is ambiguous. I found another open-source project with the same dilemma.
https://github.com/VirtualGL
https://virtualgl.org/About/License

And they were also doing research 
https://groups.google.com/g/wx-discuss/c/DxRq209iei0

However, I may wish to alter the behavior of wxWidgets, or wxPython (C/C++ side), or in some point in the future add other LGPL (C++) modules , that may be modified. I don’t want to have to go back and get contributors to agree with license changes.

IMHO LGPL is what widgets authors intended.

That said, now I’m wondering if the project should be split, compiled core as LGPL and Python.
Python can’t statically link; I don’t see Python that contributors/developers would, or should have any restrictions from a LGPL’d Core, for example your python tools could be MIT… right? But this is your decision 

Still stewing on this

**Reply by gswifort** [reply]
I think it will be ok.
